### DISCUSSION TITLE
EAS build fails when using hermes (android) in a project with Expo-Router

### DISCUSSION THREAD

**Discussion by seyaobey-dev** [expo/router#144] [Q&A]
I'm not sure if this is bug or something wrong on my hand. That's why I'm reporting it as a discussion hoping to find some help.

EAS build fails when **hermes** is enabled in my app (using Expo Router). But it does not fail when I'm using the version of my app without **expo-router** (classic react-navigation). 
```
[stderr] /home/expo/workingdir/build/android/app/build/generated/assets/react/release/index.android.bundle:182810:9: error: Invalid regular expression: Quantifier has nothing to repeat
[stderr]     N = /^((?<protocol>http[s]?):\/\/)?(?<host>localhost)(:(?<port>\d+))?$/;
```

There is an related issue in [facebook hermes repo](https://github.com/facebook/hermes/issues/46). It looks like hermes has issues with regex [see here](https://github.com/facebook/hermes/blob/main/doc/RegExp.md).

But then why the same project is building just fine in the version without `expo-router`?
(I'm willing to share the link to the repo or the eas build logs/account for further investigations. I hope this discussion find some visibility)

**Comment by EvanBacon** [comment]
Expo Router uses a lot of regex as I am a regex-absolutist. But I imagine this is a Hermes issue. If I were to guess, it's a lack of support for the named groups `?<host>` which aren't directly used in Expo Router but perhaps some dependency(?)

**Reply by seyaobey-dev** [reply]
@EvanBacon Thanks for the reply. FWIW, I seem to have fixed the issue by upgrading `metro` and `metro-source-map` dependencies to `0.73.6`.
